ping-thing-client.mjs

```diff
@@ -116,35 +118,37 @@ async function pingThing() {
 
         console.log(`Sending ${signature}`);
 
-        while (true) {
-          try {
-            slotSent = await getNextSlot();
-
-            txStart = Date.now();
-
-            await safeRace([
-              mSendAndConfirmTransaction(transactionSignedWithFeePayer, {
-                commitment: "confirmed",
-                maxRetries: 0n,
-                skipPreflight: true,
-              }),
-              timeout(TX_RETRY_INTERVAL * txSendAttempts),
-            ]);
-
-            console.log(`Confirmed tx ${signature}`);
-
-            break;
-          } catch (e) {
-            if (e.message === "Timeout") {
-              console.log(
-                `Tx not confirmed after ${TX_RETRY_INTERVAL * txSendAttempts++
-                }ms, resending`
-              );
+        let sendAbortController;
+        function sendTransaction() {
+          sendAbortController = new AbortController()
+          mSendTransactionWithoutConfirming(transactionSignedWithFeePayer, {
+            abortSignal: sendAbortController.signal,
+            commitment: COMMITMENT_LEVEL,
+            maxRetries: 0n,
+            skipPreflight: true,
+          }).catch(e => {
+            if (e instanceof Error && e.name === 'AbortError') {
+              return;
             } else {
               throw e;
             }
-          }
+          });
         }
+        slotSent = await getNextSlot();
+        const sendRetryInterval = setInterval(() => {
+          sendAbortController.abort();
+          console.log(`Tx not confirmed after ${TX_RETRY_INTERVAL * txSendAttempts++}ms, resending`);
+          sendTransaction();
+        }, TX_RETRY_INTERVAL);
+        txStart = Date.now();
+        sendTransaction();
+        await mConfirmRecentSignature({
+          abortSignal: AbortSignal.any([]),
+          commitment: COMMITMENT_LEVEL,
+          signature,
+        });
```

Comment:
OK, so I get what your original implementation was angling at now @WilfredAlmeida. You want to:

1. Get the slot.
2. Fire off the transaction
3. Start _one_ confirmer
4. Retry the transaction send if you haven't gotten confirmation in x milliseconds.

Reply:
Yes that's correct